### PR TITLE
chore: merge configs in TAV file and update versions

### PR DESCRIPTION
**Pull request by david-luna**
PR to simplify a bit the `.tav.yml` file and also tun the version to have faster CI builds.
- some entries are merged since they were specific for a nodejs version lower than the supported (v14.17). See `got` as an example
- some entries have now a `update-versions` config with strategy `latest-minors` to reduce the number of versions tested. See `undici` as an example.

### Checklist

- [ ] Implement code
- [ ] Add tests
- [ ] Update TypeScript typings
- [ ] Update documentation
- [ ] Add CHANGELOG.asciidoc entry
- [x] Commit message follows [commit guidelines](https://github.com/elastic/apm-agent-nodejs/blob/main/CONTRIBUTING.md#commit-message-guidelines)

### CODE REVIEW

File: dev-utils/ci-tav-slow-jobs.sh [hunk 1]
```diff
@@ -19,6 +19,12 @@ fi
 set -o errexit
 set -o pipefail
 
+# Make sure we have json package
+jsonPath=$(which json)
```
Comment (david-luna) [inline dev-utils/ci-tav-slow-jobs.sh:23]: Note to reviewer: sometimes I find myself switching to a different node version and the `json` package is not present.

File: .tav.yml [hunk 2]
```diff
@@ -102,15 +89,13 @@ ioredis-v2-v4:
 ioredis-v4-v5:
   name: ioredis
   versions: '4.0.0 || 4.0.2 || 4.1.0 || 4.2.3 || 4.3.1 || 4.4.0 || 4.5.1 || 4.6.3 || 4.7.0 || 4.8.0 || 4.9.5 || 4.10.4 || 4.11.2 || 4.12.2 || 4.13.1 || 4.14.4 || 4.15.1 || 4.16.3 || 4.17.3 || 4.18.0 || 4.19.4 || 4.20.0 || 4.21.0 || 4.22.0 || 4.23.1 || 4.24.6 || 4.25.0 || 4.26.0 || 4.27.11 || ^4.28.5'
-  node: '>=6'
   commands: node test/instrumentation/modules/ioredis.test.js
   update-versions:
     mode: latest-minors
     include: '>=4.0.0 <5'
```
Comment (trentm) [inline .tav.yml:95]: nit (fine for a separate PR): Perhaps `mode: max-5` for ioredis@4. That's a lot of latest-minors that feels like more than we need to continually test.

File: .tav.yml [hunk 3]
```diff
@@ -153,15 +138,13 @@ mongodb-core:
...
-  commands: node test/instrumentation/modules/mongodb/mongodb.test.js
-mongodb-4:
+mongodb-3-4:
   name: mongodb
-  versions: '>=4 <5'
-  node: '>=12.9.0'
+  versions: '3.3.0 || 3.3.5 || 3.4.1 || 3.5.11 || 3.6.12 || 3.7.4 || 4.0.1 || 4.1.4 || 4.2.2 || 4.3.1 || 4.4.1 || 4.5.0 || 4.6.0 || 4.7.0 || 4.8.1 || 4.9.1 || 4.10.0 || 4.11.0 || 4.12.1 || 4.13.0 || 4.14.0 || 4.15.0 || 4.16.0 || ^4.17.1'
   commands: node test/instrumentation/modules/mongodb/mongodb.test.js
+  update-versions:
+    mode: latest-minors
```
Comment (trentm) [inline .tav.yml:146]: nit: Similar to the above comment. Consider `mode: max-5` for this group.

File: .tav.yml [hunk 4]
```diff
@@ -606,7 +581,9 @@ aws-sdk:
 #   in the 'undici:request:create' diagnostic message.
 undici:
   name: undici
-  versions: '>=4.7.1 <6'
+  versions: '4.7.1 || 4.7.3 || 4.8.2 || 4.9.5 || 4.10.4 || 4.11.3 || 4.12.2 || 4.13.0 || 4.14.1 || 4.15.1 || 4.16.0 || 5.0.0 || 5.1.1 || 5.2.0 || 5.3.0 || 5.4.0 || 5.5.1 || 5.6.1 || 5.7.0 || 5.8.2 || 5.9.1 || 5.10.0 || 5.11.0 || 5.12.0 || 5.13.0 || 5.14.0 || 5.15.2 || 5.16.0 || 5.17.1 || 5.18.0 || 5.19.1 || 5.20.0 || 5.21.2 || 5.22.1 || 5.23.0 || 5.24.0 || 5.25.4 || 5.26.5 || 5.27.2 || ^5.28.1'
   commands: node test/instrumentation/modules/undici/undici.test.js
-  node: '>=14'
+  update-versions:
+    mode: latest-minors
```
Comment (trentm) [inline .tav.yml:587]: nit: Similar to above. Consider 'max-5' for this.

File: dev-utils/ci-tav-slow-jobs.sh [hunk 5]
```diff
@@ -19,6 +19,12 @@ fi
...
+if [ "$jsonPath" == "" ]; then
+    npm i -g json
```
Comment (trentm) [inline dev-utils/ci-tav-slow-jobs.sh:25]: I *really* don't like a script making global changes to my system as a side-effect. I would prefer we just cope with the occasional failure, or how about we just add `json.js` to our dev-utils/ area. It is a single JS file?

File: .tav.yml (hunk 2 above)
Comment (david-luna) [reply .tav.yml]: done!

File: .tav.yml (hunk 3 above)
Comment (david-luna) [reply .tav.yml]: done!

**Comment by trentm** [comment]
@elasticmachine, run elasticsearch-ci/docs

**Review by trentm — APPROVED** [review]

**Pull request merged**